bio_hansel/subtyper.py

```diff
@@ -437,6 +438,37 @@ def absent_downstream_subtypes(subtype: str, subtypes: pd.Series, scheme_subtype
     return absentees if len(absentees) > 0 else None
 
 
+def missing_nests(st: Subtype, subtype: str, df_positive: pd.DataFrame) -> Subtype:
+    """Find nested subtypes that are missing from the final subtype call
+
+    Args:
+        subtype: Final subtype result
+        positive_subtypes: List of unique positive subtypes found
+    
+    Returns:
+        List of missing hierarchical subtypes or `None` if there are no missing nested hierarchical subtypes.
+    """
+
+    pos_subtypes_str = [x for x in df_positive.subtype.unique()]
+    broken_subs = re.compile(r'^\d(\d*\.+\d+)*$')
+    primary_subtypes = []
+    for value in range(len(subtype)+1):
+        x = subtype[0:value]
+        z = re.search(broken_subs, x)
+        if z:
+            forms = primary_subtypes.append(x)
+
+    d = {key: 'a' for (key) in primary_subtypes}
+
+    for subs in pos_subtypes_str:
+        try:
+            del d[subs]
+        except:
```

Comment:
Should be explicitly catching the exception:

```python
except KeyError:
    pass
```

Reply:
Unnecessary now